teos/teosd.py

```diff
@@ -198,16 +199,23 @@ def main(config):
             # FIXME: 92-block-data-during-bootstrap-db
             chain_monitor.monitor_chain()
 
-            # Start the API (using gunicorn) and the RPC server
-            # FIXME: We may like to add workers depending on a config value
-            subprocess.Popen(
-                [
-                    "gunicorn",
-                    f"--bind={config.get('API_BIND')}:{config.get('API_PORT')}",
-                    f"teos.api:serve(internal_api_endpoint='{INTERNAL_API_ENDPOINT}', "
-                    f"min_to_self_delay='{config.get('MIN_TO_SELF_DELAY')}', log_file='{config.get('LOG_FILE')}')",
-                ]
-            )
+            # Start the API and the RPC server
+            api_endpoint = f"{config.get('API_BIND')}:{config.get('API_PORT')}"
+            if config.get("WSGI") == "gunicorn":
+                # FIXME: We may like to add workers depending on a config value
+                subprocess.Popen(
+                    [
+                        "gunicorn",
+                        f"--bind={api_endpoint}",
+                        f"teos.api:serve(internal_api_endpoint='{INTERNAL_API_ENDPOINT}', "
+                        f"endpoint='{api_endpoint}', min_to_self_delay='{config.get('MIN_TO_SELF_DELAY')}', "
+                        f"log_file='{config.get('LOG_FILE')}')",
+                    ]
+                )
+            else:
+                os.environ["WERKZEUG_RUN_MAIN"] = "true"
```

Comment:
Is it necessary to suppress the warning? After all, it's giving good advice.

Reply:
Fixed in 54e7f24e79d8e9497fe543cfea839b6c7677453e